zrml/simple-disputes/src/lib.rs

```diff
@@ -94,22 +172,97 @@ mod pallet {
         InvalidMarketStatus,
         /// On dispute or resolution, someone tried to pass a non-simple-disputes market type
         MarketDoesNotHaveSimpleDisputesMechanism,
+        StorageOverflow,
+        OutcomeMismatch,
+        CannotDisputeSameOutcome,
+        MarketIsNotReported,
+        /// The maximum number of disputes has been reached.
+        MaxDisputesReached,
     }
 
-    #[pallet::event]
-    pub enum Event<T>
-    where
-        T: Config, {}
-
     #[pallet::hooks]
     impl<T: Config> Hooks<T::BlockNumber> for Pallet<T> {}
 
-    impl<T> Pallet<T>
-    where
-        T: Config,
-    {
+    #[pallet::call]
+    impl<T: Config> Pallet<T> {
+        #[pallet::weight(T::WeightInfo::reserve_outcome(
+            T::MaxDisputes::get(),
+            CacheSize::get(),
+            CacheSize::get(),
+        ))]
+        #[transactional]
+        pub fn reserve_outcome(
```

Comment:
Why is this called `reserve_outcome`? I mean... you're not reserving the outcome when you're calling this, are you? Not sure what to call this though? `vouch`?

Reply:
We use `T::AssetManager::reserve_named` in there. And you have an outcome suggestion as a dispatch function parameter.

Yeah sure the naming might be misleading here because the same outcome can be chosen two or more times. So it's not really a form of reservation.

`vouch` sounds okay to me. Or we could even say `suggest`

Reply:
5d873d218d42005cfc97e18168da4d846ad55bf5